pkgs/native_assets_cli/lib/src/validation.dart

```diff
@@ -6,149 +6,163 @@ import 'dart:io';
 
 import 'config.dart';
 import 'encoded_asset.dart';
+import 'extension.dart';
 import 'hooks/syntax.g.dart';
 
 typedef ValidationErrors = List<String>;
 
-Future<ValidationErrors> validateBuildInput(BuildInput input) async {
-  final syntaxErrors = BuildInputSyntax.fromJson(input.json).validate();
-  if (syntaxErrors.isNotEmpty) {
-    return [...syntaxErrors, _semanticValidationSkippedMessage];
+/// The base protocol for `hook/build.dart` and `hook/link.dart` which can be
+/// extended with [ProtocolExtension]s.
+///
+/// This class currently contains the logic of the protocol which is similar to
+/// the protocol extensions: validation rules.
```

Comment:
nit: it's not super clear what "similar" means here? Is this logic that is shared by all/most protocol extensions?